docs/adr/0001-implement-nested-facets-tagging-for-specialist-documents-as-a-single-dropdown.md

```diff
@@ -0,0 +1,71 @@
+
+Date: 2025-02-14
+
+## Status
+
+Proposed
+
+## Related PRs
+
+- [Implemented as part of this PR](https://github.com/alphagov/specialist-publisher/pull/2923)
+
+## Context
+
+Nested facets are required by a department as part of their new finder. Their existing finder contains: 
+
+- For filtering, a "Facet" dropdown and a "Sub facet" dropdown to filter for documents. Users are able to filter through a single facet value lookup at a time. 
+- The documents are tagged with multiple Facet and Sub facet values
+
+We need to be able to replicate this behaviour with minimal efforts to meet a deadline. 
+
+## Decision
+
+What we have implemented:
+
+- On the publisher side for tagging, a single dropdown that merges the facets' values (main facet and sub facets, essentially leaf nodes)
+- When we send the document to Publishing API, we write logic into the document model to figure out whether the leaf node is a parent facet or sub-facet value in the dropdown, and split the metadata we send to Publishing API into its own facet and a separate sub-facet in `specialist-publisher/app/models/document.rb` 
+  - Reasons for this being if we send them as separate facets, we can keep Publishing API and Search API functionality unchanged. 
+- When we load the edit page or new page again, the document is loaded from Publishing API, with the facet and sub-facet being separate facets, but the dropdown element expects a combined leaf node dropdown as mentioned in the first point, so we have to merge them back together again into the main facet field in `specialist-publisher/app/controllers/documents_controller.rb`.
+
+## Consequences
+
+Pros: 
+
+- Restrict user input to only valid options with a single dropdown. 
+- This supports both multiple select entries and single select entries. 
+- The workflow discussed as part of analysis document works.
+
+Cons:
+
+- Logic to merge the parent facet values, sub-facet values at form phase, split them at send phase, merge them again at load phase might be a little convoluted.
+
+## Alternatives proposed
+
+### Have tagging done as 2 separate facets with validation 
+
+Even on publisher side, surface all of a facet's parents facets as one dropdown, and all of the sub-facets as a separate dropdown. As outlined on [this branch](https://github.com/alphagov/specialist-publisher/compare/intellectual-property-office-finder_2?expand=1).
+
+Pros:
+
+- Consistency of 2 separate facets being displayed to users upon tagging and viewing metadata. 
+- No change logic required to split or merge facets
+
+Cons:
+
+- Repeat of information. The sub-facets are also prefixed with parent facet in any case (e.g. to differentiate between multiple "Not Applicable" sub-facet entries). Users will have to tag both parent facet and the sub-facet with the prefix. 
+- There is no structural tie between the parent facet and the sub-facet when tagging. Nothing stopping the user from tagging mismatching pairs or not tagging a sub-facet. 
```

Comment:
"Not tagging a sub-facet" is allowed though, isn't it? As per our [assumptions](https://docs.google.com/document/d/1GaSnboJ8_0KnCd3uaCDxuJg0JtECh45hOfLnuoDkaAI/edit?tab=t.0#heading=h.cpq4r1qiln6j).

Reply:
hmm I'm not sure we clarified between 
- Having a mixture of leaf nodes - hence tagging a main-facet because it doesn't have sub-facets
- Tagging a main-facet only despite it also having sub-facets

I think in the future we do want to support tagging main-facet only despite having sub-facets, but we felt like we could clarify that later down the line, as the current MVP doesn't require it.